app/services/iiif_print/manifest_builder_service_behavior.rb

```diff
@@ -0,0 +1,104 @@
+module IiifPrint
+  module ManifestBuilderServiceBehavior
+    def initialize(*args,
+                   version: IiifPrint.config.default_iiif_manifest_version,
+                   iiif_manifest_factory: iiif_manifest_factory_for(version),
+                   &block)
+      super(*args, iiif_manifest_factory: iiif_manifest_factory, &block)
+      @version = version.to_i
+      @presenter_solr_docs = {}
+    end
+
+    def manifest_for(presenter:)
+      build_manifest(presenter: presenter)
+    end
+
+    private
+
+    VERSION_TO_MANIFEST_FACTORY_MAP = {
+      2 => ::IIIFManifest::ManifestFactory,
+      3 => ::IIIFManifest::V3::ManifestFactory
+    }.freeze
+
+    def iiif_manifest_factory_for(version)
+      VERSION_TO_MANIFEST_FACTORY_MAP.fetch(version.to_i)
+    end
+
+    ##
+    # Allows for the display of metadata for child works in UV
+    #
+    # @see https://github.com/samvera/hyrax/blob/main/app/services/hyrax/manifest_builder_service.rb
+    def build_manifest(presenter:)
+      # ::IIIFManifest::ManifestBuilder#to_h returns a
+      # IIIFManifest::ManifestBuilder::IIIFManifest, not a Hash.
+      # to get a Hash, we have to call its #to_json, then parse.
+      #
+      # wild times. maybe there's a better way to do this with the
+      # ManifestFactory interface?
+      manifest = manifest_factory.new(presenter).to_h
+      hash = JSON.parse(manifest.to_json)
+      hash = send("sanitize_v#{@version}", hash: hash, presenter: presenter)
+      send("sorted_canvases_v#{@version}", hash: hash, sort_field: IiifPrint.config.sort_iiif_manifest_canvases_by)
+    end
+
+    def sanitize_v2(hash:, presenter:)
+      hash['label'] = CGI.unescapeHTML(sanitize_value(hash['label'])) if hash.key?('label')
+      hash.delete('description') # removes default description since it's in the metadata fields
+      hash['sequences']&.each do |sequence|
+        sequence['canvases']&.each do |canvas|
+          canvas['label'] = CGI.unescapeHTML(sanitize_value(canvas['label']))
+          apply_v2_metadata_to_canvas(canvas: canvas, presenter: presenter)
+        end
+      end
+      hash
+    end
+
+    def sanitize_v3(hash:, presenter:)
+      # TODO: flesh out metadata for v3
```

Comment:
Is there a ticket for v3 updates?

Reply:
Oh I don't think so actually, nor for AllinsonFlex